modules/__tests__/describePopState.js

```diff
@@ -18,6 +18,14 @@ const describePopState = (createHistory) => {
 
       window.history.back()
     })
+
+    it('is called when only a hash is navigated to', (done) => {
+      unlisten = history.listenBefore(() => {
+        done()
```

Comment:
I don't think that's unnecessary. Technically these are two separate conditions anyway – one to check that we get correct `listenBefore` hook semantics for navigation actions triggered outside of `history`; yours is just to make sure that for a normal `listen` listener, we do fire the callback and we don't fire it twice.